cli/src/main/java/com/devonfw/tools/ide/tool/plugin/PluginBasedCommandlet.java

```diff
@@ -93,41 +95,124 @@ public Path getPluginsInstallationPath() {
-      fileAccess.mkdirs(pluginsInstallationPath);
-      installPlugins(getPlugins().getPlugins());
+      if (this.context.getIdeHome() != null) {
+        List<Path> markerFiles = fileAccess.listChildren(this.context.getIdeHome().resolve(".ide"), Files::isRegularFile);
+        for (Path path : markerFiles) {
+          if (path.getFileName().toString().startsWith("plugin." + getName())) {
+            this.context.debug("Plugin marker file {} got deleted.", path);
+            fileAccess.delete(path);
+          }
+        }
+      }
+
+      createEditionMarkerFile();
+    }
+    fileAccess.mkdirs(pluginsInstallationPath);
+    installPlugins(pc);
+  }
+
+  private void installPlugins(ProcessContext pc) {
+    installPlugins(getPlugins().getPlugins(), pc);
+  }
+
+  /**
+   * Creates a marker file for an installed edition.
+   */
+  public void createEditionMarkerFile() {
+    if (this.context.getPluginsPath() != null) {
+      Path pluginsPath = retrieveEditionMarkerFilePath(getName());
+      FileAccess fileAccess = this.context.getFileAccess();
+      fileAccess.mkdirs(this.context.getPluginsPath().resolve(getName()));
+      fileAccess.touch(pluginsPath);
+    }
+  }
+
+  /**
+   * @param toolName the tool name to search for.
+   * @return Path to the edition marker file.
+   */
+  public Path retrieveEditionMarkerFilePath(String toolName) {
+    if (this.context.getPluginsPath() != null) {
+      return this.context.getPluginsPath().resolve(toolName).resolve("." + getInstalledEdition());
     }
-    // else ? check all plugins are installed, retry failed plugin installations ?
+    return null;
   }
 
   /**
    * Method to install active plugins or to handle install for inactive plugins
    *
    * @param plugins as {@link Collection} of plugins to install.
+   * @param pc the {@link ProcessContext} to use.
    */
-  protected void installPlugins(Collection<ToolPluginDescriptor> plugins) {
+  protected void installPlugins(Collection<ToolPluginDescriptor> plugins, ProcessContext pc) {
     for (ToolPluginDescriptor plugin : plugins) {
       if (plugin.active()) {
-        try (Step step = this.context.newStep("Install plugin " + plugin.name())) {
-          installPlugin(plugin, step);
+        if (retrievePluginMarkerFilePath(plugin) != null && Files.exists(retrievePluginMarkerFilePath(plugin))) {
+          this.context.debug("Markerfile for IDE: {} and active plugin: {} already exists.", getName(), plugin.name());
+        } else {
+          try (Step step = this.context.newStep("Install plugin " + plugin.name())) {
+            installPlugin(plugin, step, pc);
+            createPluginMarkerFile(plugin);
+          }
         }
       } else {
-        handleInstall4InactivePlugin(plugin);
+        if (retrievePluginMarkerFilePath(plugin) != null && Files.exists(retrievePluginMarkerFilePath(plugin))) {
+          this.context.debug("Markerfile for IDE: {} and inactive plugin: {} already exists.", getName(), plugin.name());
```

Comment:
Nice to have:
these two lines are identical to lines 158 and 159 except for the inactive/active word in the log message.
We could write the same code less redundant...